lib/AL_USDMaya/AL/usdmaya/nodes/TransformationMatrix.cpp

```diff
@@ -116,7 +116,7 @@ void TransformationMatrix::setPrim(const UsdPrim& prim)
     m_xform = UsdGeomXform();
   }
   m_time = UsdTimeCode(UsdTimeCode::Default());
-  m_flags = 0;
+  m_flags &= kPreservationMask;
```

Comment:
I think this should stay = 0, there is some code that will inspect the prim and set the correct flags, I hope?

Reply:
For most of the flags, yes, they are read from the prim - however, there are two flags, kPushToPrimEnabled and kReadAnimated, that are controlled not by the prim, but by attributes on the node. These should NOT be reset when the prim is reset, thus the need for this flag.

I document this above the definition of kPreservationMask, but I'll copy that here for clarity.